### PR TITLE
Add documentation for Farms

### PR DESCRIPTION
**Pull request by gabrieltron**
This PR is to add documentation for Farms. While we were working on it, we noticed that we still need to implement multi-tenancy on Farm and Growers before it's safe to make the documentation available. Here's the work on the documentation so far, it'll be improved once multi-tenancy is implemented.

### CODE REVIEW

File: docs/field_boundaries_endpoints.md [hunk 1]
```diff
@@ -61,6 +61,29 @@ POST   /users/{leafUserId}/fields/{id}/integration
 DELETE /users/{leafUserId}/fields/{id}
 ```
 
+#### Farm Resource
+
+Form of a Farm:
+```json
```
Comment (ramnasidharta) [inline docs/field_boundaries_endpoints.md:67]: I think that for reasons of markdown syntax it's better or necessary to have one blank line here
```suggestion
Form of a Farm:

```json
```

File: docs/field_boundaries_endpoints.md [hunk 2]
```diff
@@ -694,6 +717,376 @@ A JSON in the followin format.
...
+  ]
+}>
+  <TabItem value="js">
+
+  ```js
+  const axios = require('axios')
+  const TOKEN = 'YOUR_TOKEN'
+
+  const endpoint ='https://api.withleaf.io/services/fields/api/farms'
+  const headers = { 'Authorization': `Bearer ${TOKEN}` }
```
Comment (ramnasidharta) [inline docs/field_boundaries_endpoints.md:748]:
```suggestion
  const headers = { 'Authorization': 'Bearer ${TOKEN}' }
```

Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:748]: Same in other places

File: docs/field_boundaries_endpoints.md [hunk 3]
```diff
@@ -694,6 +717,376 @@ A JSON in the followin format.
...
+      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+### `POST users/{leafUserId}/farms`
+
+Creates a Farm. The request's body can provide values for the farm
```
Comment (ramnasidharta) [inline docs/field_boundaries_endpoints.md:929]:
```suggestion
Creates a Farm. The request's body provides values for the farm
```

**Review by ramnasidharta — APPROVED** [review]
Well done, mate! (Approved. Please consider my comments and note that you commited `.idea` files, those can be ignored)

File: docs/field_boundaries_endpoints.md [hunk 4]
```diff
@@ -61,6 +61,29 @@ POST   /users/{leafUserId}/fields/{id}/integration
...
 
+#### Farm Resource
+
+Form of a Farm:
+```json
+{
+  "id": "long",
+  "name": "str",
+  "providerName": "str",
+  "providerFarmId": "UUID",
```
Comment (GMBueno) [inline docs/field_boundaries_endpoints.md:72]: can we remove this from the response? It's unnecessary, from my perspective....

Comment (GMBueno) [reply docs/field_boundaries_endpoints.md:72]: I am referring to "providerFarmId", this is the id of the farm on the provider right?

File: docs/field_boundaries_endpoints.md [hunk 5]
```diff
@@ -61,6 +61,29 @@ POST   /users/{leafUserId}/fields/{id}/integration
...
 ```
 
+#### Farm Resource
+
+Form of a Farm:
+```json
+{
+  "id": "long",
+  "name": "str",
+  "providerName": "str",
```
Comment (GMBueno) [inline docs/field_boundaries_endpoints.md:71]: we have to define if the convention will be "providerName" or "provider"... we have both now

File: docs/field_boundaries_endpoints.md (hunk 4 above)
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:72]: Yes, it is. Why unnecessary? datasync can create farms, so they will have this attribute set

File: docs/field_boundaries_endpoints.md (hunk 5 above)
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:71]: Yep. I vote for `provider`

**Review by GMBueno — CHANGES_REQUESTED** [review]
good job! I added a few comments/questions I would like to know before approving 😄

File: docs/field_boundaries_endpoints.md [hunk 6]
```diff
@@ -694,6 +717,376 @@ A JSON in the followin format.
...
+    "fieldIds": ["00000000-0000-0000-0000-000000000000"],
+    "growerId": 123
+  }
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+### `DELETE users/{leafUserId}/farms/{id}`
```
[the remark this answers is not in the document]
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:874]: "Only at leaft" for the 1st question. "Maybe" for the 2nd. If that farm was created from pooling then yes, we do will create it again. We know 0 real use case for datasync, so I think documenting this kind of endpoint are 100% a product decision. The fact is that all endpoint that allows writting are conflicting with datasync. Currently we're simply ignoring this fact.

File: docs/field_boundaries_endpoints.md [hunk 7]
```diff
@@ -694,6 +717,376 @@ A JSON in the followin format.
...
+  curl -X DELETE \
+      -H 'Authorization: Bearer YOUR_TOKEN' \
+      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms/{id}'
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+### `POST users/{leafUserId}/farms`
```
[the remark this answers is not in the document]
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:927]: Nope, although we can add a farm to a field on field creation. Does this functionality you're asking about make sense?

File: docs/field_boundaries_endpoints.md [hunk 8]
```diff
@@ -694,6 +717,376 @@ A JSON in the followin format.
...
+      -H 'Authorization: Bearer YOUR_TOKEN' \
+      -d '{ "name": "name", "growerId": 12345 }'
+      'https://api.withleaf.io/services/fields/api/users/{leafUserId}/farms'
+  ```
+
+  </TabItem>
+</Tabs>
+
+
+### `PUT users/{leafUserId}/farms`
```
[the remark this answers is not in the document]
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md:1007]: Nope

File: docs/field_boundaries_endpoints.md [hunk 9]
```diff
@@ -118,7 +164,7 @@ A JSON array containing Fields.
   const TOKEN = 'YOUR_TOKEN'
 
   const endpoint ='https://api.withleaf.io/services/fields/api/fields'
-  const headers = { 'Authorization': `Bearer ${TOKEN}` }
```
Comment (meyer1994) [inline docs/field_boundaries_endpoints.md:121]: Why have you changed this? It is valid JavaScript

**Review by meyer1994 — CHANGES_REQUESTED** [review]

File: docs/field_boundaries_endpoints.md (hunk 9 above)
Comment (ramnasidharta) [reply docs/field_boundaries_endpoints.md]: Approve it

**Review by meyer1994 — APPROVED** [review]

**Review by GMBueno — APPROVED** [review]

**Pull request merged**